pkg/daemon/ceph/osd/volume.go

```diff
@@ -721,8 +762,32 @@ func GetCephVolumeRawOSDs(context *clusterd.Context, clusterInfo *client.Cluster
 			CVMode:        cvMode,
 			Store:         "bluestore",
 		}
-		osds = append(osds, osd)
 
+		// Get encrypted key
+		if dmCryptKeyValue != "" {
+			// Add the encryption key name of the OSDInfo
+			osd.EncryptedSecretName = generateOSDEncryptionSecretName(osdID)
+
+			// Generate the encrypted key Kubernetes Secret
+			s := generateOSDEncryptedKeySecret(osdID, clusterInfo.Namespace, dmCryptKeyValue)
+
+			// Set the ownerref to the Secret
+			k8sutil.SetOwnerRef(&s.ObjectMeta, &clusterInfo.OwnerRef)
+
+			// Create the Kubernetes Secret
+			_, err := context.Clientset.CoreV1().Secrets(clusterInfo.Namespace).Create(s)
+			if err != nil {
```

Comment:
Since the prepare job is skipped for a device set, I don't think this can ever happen. If it exists somehow, this should be a failure anyway, if we replace an OSD, I believe we should re-create an encryption key (just for safety).

Reply:
Checking now since the code moved to the operator.